File: URI/C/JornadaNasEstrelas.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
typedef unsigned int uint;
typedef unsigned long long int ulli;
#define true 1
#define false 0

typedef struct infos_fazenda
{
    long long int carneiros;
    int visitada;
} fazenda;


int main(void){
    long long int carneiros_t = 0, n, farms_afetadas = 0;
    scanf("%lld", &n);
    fazenda farm[n];

    for (int i = 0; i < n; i++)
    {
        scanf("%lld", &farm[i].carneiros);
        carneiros_t += farm[i].carneiros;
        farm[i].visitada = 0;
    }
    int indice = 0, prox_indice = 0;
    while (indice != -1 && indice != n)
    {
        // printf("estrela = %d %d\n", indice+1, farm[indice].visitada);
        if (farm[indice].carneiros%2 == 0){
            prox_indice--;
        }
        else
        {
            prox_indice++;
        }

        if (farm[indice].carneiros > 0)
        {
            farm[indice].carneiros--;
            carneiros_t--;
        }

        if (farm[indice].visitada == false)
        {
            farm[indice].visitada = true;
            farms_afetadas++;
        }
        indice = prox_indice;

    }
    
    printf("%lld %lld\n", farms_afetadas ,carneiros_t);
    
    return 0;
}

File: URI/Python/JornadaNasEstrelas.py
class fazenda():
    def __init__(self, carneiros):
        self.carneiros = carneiros
        self.visitada = False

n = int(input())
farm = [fazenda(int(x)) for x in input().split()]

for x in farm:
    carneiros_t+= x.carneiros
farms_afetadas = 0
indice = 0
prox_indice = 0

while indice != -1 and indice != n:
    if farm[indice].carneiros % 2 == 0:
        prox_indice-=1
    else:
        prox_indice+=1
    
    if farm[indice].carneiros > 0:
        farm[indice].carneiros-=1
        carneiros_t-=1
    
    if farm[indice].visitada == False:
        farm[indice].visitada = True
        farms_afetadas +=1
    
    indice = prox_indice

print("{} {}".format(farms_afetadas, carneiros_t))